app/src/main/java/com/a0xffffffff/dinesum/RequestState.java

```diff
@@ -1,4 +1,24 @@
 package com.a0xffffffff.dinesum;
 
-public class RequestState {
+abstract public class RequestState {
```

Comment:
Why do we need this instead of having a state field inside a Request object?

Reply:
I think we wanted to implement functions goNext() and goPrevious() to transition between the different states, so if we had the RequestState as a base class, then the subclasses could implement those functions